triggers/service/auth/src/main/scala/com/daml/auth/middleware/oauth2/Cli.scala

```diff
@@ -0,0 +1,268 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.auth.middleware.oauth2
+
+import akka.http.scaladsl.model.Uri
+import com.auth0.jwt.algorithms.Algorithm
+import com.daml.auth.middleware.oauth2.Config.{
+  DefaultCookieSecure,
+  DefaultHttpPort,
+  DefaultLoginTimeout,
+  DefaultMaxLoginRequests,
+}
+import com.daml.cliopts
+import com.daml.jwt.{
+  ECDSAVerifier,
+  HMAC256Verifier,
+  JwksVerifier,
+  JwtVerifierBase,
+  JwtVerifierConfigurationCli,
+  RSA256Verifier,
+}
+import com.typesafe.scalalogging.StrictLogging
+import pureconfig.{ConfigReader, ConfigSource, ConvertHelpers}
+import pureconfig.error.ConfigReaderException
+import scopt.OptionParser
+
+import java.io.File
+import pureconfig.generic.semiauto._
+
+import java.nio.file.{Path, Paths}
+import scala.concurrent.duration
+import scala.concurrent.duration.FiniteDuration
+
+sealed trait ConfigError extends Product with Serializable {
+  def msg: String
+}
+case object MissingConfigError extends ConfigError {
+  val msg = "Missing auth middleware config file"
+}
+final case class ConfigParseError(msg: String) extends ConfigError
+
+case class Cli(
```

Comment:
```suggestion
final case class Cli(
```
Unless you have a reason not to.